packages/terra-data-grid/src/FlowsheetDataGrid.jsx

```diff
@@ -227,35 +235,51 @@ function FlowsheetDataGrid(props) {
     } else if (selectedCells.current.length) {
       setCellSelectionAriaLiveMessage(intl.formatMessage({ id: 'Terra.flowsheetDataGrid.cells-selected' }, { cellCount: selectedCells.current.length }));
     }
-  }, [intl, rows, columns, setCellSelectionAriaLiveMessage]);
+  }, [intl, flowsheetRows, flowsheetSections, columns, setCellSelectionAriaLiveMessage]);
 
-  const selectCellRange = useCallback((rowIndex, columnIndex) => {
+  const selectCellRange = useCallback((rowId, columnId, sectionId) => {
     if (anchorCell.current === null) {
       return;
     }
 
-    const anchorRowIndex = rows.findIndex(row => row.id === anchorCell.current.rowId);
+    const rowsToSearch = flowsheetSections
+      ? flattenSections(flowsheetSections)
+      : rows;
```

Comment:
Instead of having to flatten the sections and try to figure out the position in the DOM, can't you just use a querySelector since you have a reference to the grid?  You know that the section id is set as an attribute on the tr element (I believe).  You can use that knowledge to find the bounds of the section.